butte-build/src/ir/transform.rs

```diff
@@ -313,6 +310,53 @@ impl<'a> Builder<'a> {
         })
     }
 
+    fn new_rpc(&mut self, rpc: &ast::Rpc<'a>, pos: usize) -> Result<bool> {
+        let ident = &self.make_fully_qualified_ident(ir::Ident::from(&rpc.id));
+
+        let mut methods = Vec::new();
+
+        for m in &rpc.methods {
+            let ident = ir::Ident::from(&m.id);
+            let snake_ident = ir::Ident::from(m.id.as_ref().to_snake_case());
+            let request_type = ir::DottedIdent::from(&m.request_type);
+            let response_type = ir::DottedIdent::from(&m.response_type);
+            let streaming = match &m.metadata {
```

Comment:
Should we parse these into enums instead of converting them during codegen? Happy to do this in a follow up.

Reply:
I think it's nice for the parsing layer to remain dumb, and for the semantic layer to pick the supported attributes into enums.

Reply:
Yeah, I go back and forth about just on _how_ dumb it should be. In this case it seems like a little more intelligence would be okay, but I also don't feel that strongly about it. I may tinker around a bit after this is merged.